### ISSUE TITLE
Serial number of a device does not support extended filters

### ISSUE THREAD

**Issue by veritas-sot** [nautobot/nautobot#5805]
### Environment
* Nautobot version (Docker tag too if applicable): v2.2.5
* Python version: 3.11
* Database platform, version: postgres
* Middleware(s): None

### Steps to Reproduce
Execute graphql query to get a list of all devices that have no serial number
``
{
  devices (serial:"") {
    name
    serial
  }
}
``

For most string filters there’s a __nre (not regex) variant, which could do something like name__nre=".+", but looks like serial doesn’t have any of the variants enabled.

### Expected Behavior
To get a list of all devices with no serial number.

### Observed Behavior
I got a list of all devices

**Comment by HanlinMiao** [comment]
The `serial` filter field on `DeviceFilterSet` is denoted as `serial = MultiValueCharFilter(lookup_expr="iexact")`. It should be a good starting point to investigate whether `iexact` fits this use case here.

**Comment by glennmatthews** [comment]
Yeah, I think because we're explicitly specifying the `lookup_expr` in the filter definition, it's not getting the auto-added lookup variants `__n`, `__re`, and so forth. Probably need a generic fix somewhere.

**Comment by lampwins** [comment]
When `lookup_expr` is present, treat it as the default, but still generate all of the other variants.
